Extensions/PrimitiveDrawing/Extension.cpp

```diff
@@ -168,9 +187,194 @@ void DeclarePrimitiveDrawingExtension(gd::PlatformExtension& extension) {
+                _("drawPathArcTo"),
+                _("drawPathArcTo"
+                " with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/arc24.png",
+                "res/actions/arc.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("X control point"))
+      .AddParameter("expression", _("Y control point"))
+      .AddParameter("expression", _("X coordinate end arc"))
+      .AddParameter("expression", _("Y coordinate end arc"))
+      .AddParameter("expression", _("Radius (in pixels)"))
+      .SetFunctionName("drawPathArcTo")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+*/
+
+  obj.AddAction("PathArc",
+                _("Path arc to"),
+                _("Add to a path an arc to a position, the origin come from the previous action or from BeginFillPath or PathMoveTo, by default if nothing exist, start position will be the object's position."),
+                _("Add to a path an arc at the position _PARAM1_;_PARAM2_ (radius: _PARAM3_,start angle: _PARAM4_, end angles: _PARAM5_,anticlockwise: _PARAM6_ "
+                "with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/arc24.png",
+                "res/actions/arc.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("Center x of circle"))
+      .AddParameter("expression", _("Center y of circle"))
+      .AddParameter("expression", _("Radius (in pixels)"))
+      .AddParameter("expression", _("Start angle"))
+      .AddParameter("expression", _("End angle"))
+      .AddParameter("yesorno", _("Anticlockwise"))
+      .SetFunctionName("drawPathArc")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("PathQuadraticCurveTo",
+                _("Path quadratic curve to"),
+                _("Add to a path a quadratic curve to a position, the origin come from the previous action or from BeginFillPath or PathMoveTo, by default if nothing exist, start position will be the object's position."),
+                _("Add to a path a quadratic curve to the position _PARAM3_;_PARAM4_ (control point: _PARAM1_;_PARAM2_) "
+                "with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("Control point x"))
+      .AddParameter("expression", _("Control point y"))
+      .AddParameter("expression", _("Destination point x"))
+      .AddParameter("expression", _("Destination point y"))
+      .SetFunctionName("drawPathQuadraticCurveTo")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("closePath",
+                _("Close Path"),
+                _("Close the path of advanced shape. This close the outline between the last point and first."),
```

Comment:
"an advanced shape". This is the first time we're mentioning shape. I wonder if we should mention in other actions a "shape" rather than a "fill path" (which seems weird?)

Reply:
i prefer talk about "shape".
"Path" is used by pixiJS but instead "path" we can talk "outlines"
and keep "fill" for filling the shape.

Reply:
`This closes the outline between the last and the first point.`

Reply:
Wend1go comment sounds good.

Reply:
And agree about using `shape`, `outline` and keeping `fill`.